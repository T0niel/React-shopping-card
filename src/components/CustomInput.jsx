import { useState } from 'react';
import PropTypes from 'prop-types';

export default function CustomInput({onChange, placeholder, type}){
    const [value, setValue] = useState('');
    return (
        <input
          className="p-1 bg-transparent border-b-2 w-[100%] outline-none focus:border-gray-300"
          type={type}
          placeholder={placeholder}
          value={value}
          onChange={(e) => {
            const val = e.target.value;
            setValue(val);
            onChange(val);
          }}
        ></input>
    );
}

CustomInput.propTypes = {
    onChange: PropTypes.func.isRequired,
    placeholder: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired
}