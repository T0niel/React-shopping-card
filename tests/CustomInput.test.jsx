import { describe, expect, it, vi } from 'vitest';
import CustomInput from '../src/components/CustomInput';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Custom input', () => {
  it('Matches snapshot', () => {
    const { container } = render(<CustomInput />);
    expect(container).toMatchSnapshot();
  });

  it('Calls onChange when changing the input', async () => {
    const user = userEvent.setup();

    const onChange = vi.fn();
    render(<CustomInput placeholder="" onChange={onChange} type="text" />);

    const input = screen.getByRole('textbox');

    await user.type(input, 'Hello');

    expect(onChange.mock.calls.length).toEqual(5);
  });

  it('Does not call onChange when input does not change', () => {
    const onChange = vi.fn();
    render(<CustomInput placeholder="" onChange={onChange} type="text" />);
    expect(onChange.mock.calls.length).toEqual(0);
  });

  it('Changes the value of input when typing', async () => {
    const user = userEvent.setup();
    render(<CustomInput placeholder="" onChange={() => {}} type="text" />);

    const input = screen.getByRole('textbox');

    await user.type(input, 'Hello');

    expect(input.value).toEqual('Hello');
  });
});
