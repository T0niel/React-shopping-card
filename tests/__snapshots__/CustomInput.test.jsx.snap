// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Custom input > Matches snapshot 1`] = `
<div>
  <input
    class="p-1 bg-transparent border-b-2 w-[100%] outline-none focus:border-gray-300"
    value=""
  />
</div>
`;
